src/components/viewers/MultiFlatmap.vue

```diff
@@ -220,82 +220,95 @@ export default {
           await this.toggleSyncMode();
       }
     },
-    multiFlatmapReady: function () {
-      this.$refs.multiflatmap.getCurrentFlatmap().enablePanZoomEvents(true); // Use zoom events for dynamic markers
-      this.flatmapReady = true;
-      this.flatmapMarkerZoomUpdate(true);
+    multiFlatmapReady: function (flatmap) {
+      if (flatmap) {
+        flatmap.enablePanZoomEvents(true); // Use zoom events for dynamic markers
+        this.flatmapReady = true;
+        const flatmapImp = flatmap.mapImp;
+        this.flatmapMarkerZoomUpdate(true, flatmapImp);
+      }
     },
     /**
      * Function used for updating the flatmap markers.
      * It will only update the markers if zoom level has changed or
      * the force flag is true.
      */
-    flatmapMarkerZoomUpdate(force) {
+    flatmapMarkerZoomUpdate(force, flatmap) {
       if (!this.flatmapReady) return;
-      let flatmapImp = this.getFlatmapImp();
-      let currentZoom = flatmapImp.getZoom()["zoom"];
-      if (force || this.zoomLevel !== currentZoom) {
-        this.zoomLevel = currentZoom;
-        flatmapImp.clearMarkers();
-        let markers = store.state.settings.markers;
-        checkMarkersAtZoomLevel(flatmapImp, markers, this.zoomLevel);
-        this.restoreFeaturedMarkers();
+
+      let flatmapImp = flatmap;
+      if (!flatmapImp)
+        flatmapImp = this.getFlatmapImp();
+
+      if (flatmapImp) {
+        let currentZoom = flatmapImp.getZoom()["zoom"];
+        if (force || this.zoomLevel !== currentZoom) {
+          this.zoomLevel = currentZoom;
+          flatmapImp.clearMarkers();
+          let markers = store.state.settings.markers;
+          checkMarkersAtZoomLevel(flatmapImp, markers, this.zoomLevel);
+          this.restoreFeaturedMarkers(flatmapImp);
+        }
       }
     },
     getFlatmapImp: function () {
-      if (this.entry.type === "Flatmap") {
-        return this.$refs.flatmap.mapImp;
-      } else if (this.entry.type === "MultiFlatmap") {
+      if (this.entry.type === "MultiFlatmap") {
```

Comment:
Does the normal flatmap not have a viewer anymore?

Reply:
There is and is handled in Flatmap.vue. This should have been removed when I did the refactoring.